File: src/main/java/com/thx/traveljournal/trip/entity/Trip.java
package com.thx.traveljournal.trip.entity;

import com.baomidou.mybatisplus.annotation.FieldStrategy;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import com.thx.traveljournal.common.entity.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDate;

/**
 * 一次旅行，是城市、行程、预算、支出和日记的归属主体。
 *
 * <p>对应数据库表 {@code trip}，字段注释与库中的 COMMENT 保持一致。</p>
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("trip")
public class Trip extends BaseEntity {
    /** 旅行标题 */
    private String title;
    /** 前台访问用的唯一短链，只允许小写字母、数字和短横线 */
    private String slug;
    /** 旅行简介，展示在前台卡片和详情页 */
    private String summary;
    /** 旅行状态：PLANNING 规划中、ONGOING 旅行中、COMPLETED 已完成、ARCHIVED 已归档 */
    private String status;
    /** 开始日期 */
    private LocalDate startDate;
    /** 结束日期，不得早于开始日期 */
    private LocalDate endDate;
    /** 默认币种，三位大写字母代码，例如 CNY */
    private String defaultCurrency;
    /** 封面图片，指向 media_asset；图片被删除时自动置空 */
    private Long coverMediaId;
    /** 仅后台可见的内部备注，不会出现在前台 */
    private String internalNote;
    /** 旅行专属主题标识，为空表示继承全站主题 */
    @TableField(updateStrategy = FieldStrategy.ALWAYS)
    private String themeKey;
}

File: frontend/src/types/trip.ts
import type {
  CoordinateSystem,
  Decimal,
  IsoDateTimeString,
  LocalDateString,
} from './common'

export const TRIP_STATUSES = ['PLANNING', 'ONGOING', 'COMPLETED', 'ARCHIVED'] as const
export type TripStatus = (typeof TRIP_STATUSES)[number]

/** 后台的旅行实体，`/admin/trips` 直接返回 entity。 */
export interface Trip {
  id: number
  createdAt: IsoDateTimeString
  updatedAt: IsoDateTimeString
  title: string
  slug: string
  summary: string | null
  status: TripStatus
  startDate: LocalDateString
  endDate: LocalDateString
  defaultCurrency: string
  coverMediaId: number | null
  /** 只给作者看的备注，不会出现在公开端。 */
  internalNote: string | null
  themeKey: string | null
}

/** 旅行工作台概览接口的聚合结果。 */
export interface TripDashboard {
  trip: Trip
  stopCount: number
  itineraryCount: number
  draftCount: number
  publishedCount: number
  budgetTotal: Decimal
  actualExpense: Decimal
  remainingBudget: Decimal
}

/** 后台的停靠点实体。坐标按 `coordinateSystem` 解释，不要假定一定是 WGS84。 */
export interface TripStop {
  id: number
  createdAt: IsoDateTimeString
  updatedAt: IsoDateTimeString
  tripId: number
  cityName: string
  regionName: string | null
  countryName: string
  countryCode: string | null
  latitude: Decimal
  longitude: Decimal
  placeId: string | null
  formattedAddress: string | null
  adcode: string | null
  coordinateSystem: CoordinateSystem
  locationSource: string | null
  arrivalDate: LocalDateString | null
  departureDate: LocalDateString | null
  sortOrder: number
  note: string | null
}

export interface TripRequest {
  title: string
  slug: string
  summary?: string
  status: TripStatus
  startDate: LocalDateString
  endDate: LocalDateString
  defaultCurrency: string
  coverMediaId?: number | null
  internalNote?: string
  themeKey?: string | null
}

/*
 * 城市停靠点的完整表单（PUT 语义）：null 表示清空这个字段。
 *
 * 可空字段写成 `| null` 而不是只有 `?`，是为了让「作者清掉了这一项」能够被明确表达出来。
 * 后端对应的列都标了 FieldStrategy.ALWAYS，收到 null 会真的写进 NULL。
 */
export interface StopRequest {
  cityName: string
  regionName?: string
  countryName: string
  countryCode?: string
  latitude: Decimal
  longitude: Decimal
  placeId?: string | null
  formattedAddress?: string
  adcode?: string
  coordinateSystem?: CoordinateSystem
  locationSource?: string
  arrivalDate?: LocalDateString | null
  departureDate?: LocalDateString | null
  sortOrder?: number
  note?: string
}
